### PR TITLE
An attempt to sync a non-mirror repo must give 400 (Bad Request)

### PR DESCRIPTION
**Pull request by petergardfjall**
Intends to address https://github.com/go-gitea/gitea/issues/19299

### CODE REVIEW

File: integrations/api_repo_test.go [hunk 1]
```diff
@@ -405,6 +405,27 @@ func testAPIRepoMigrateConflict(t *testing.T, u *url.URL) {
 	})
 }
 
+// mirror-sync must fail with "400 (Bad Request)" when an attempt is made to
```
Comment (petergardfjall) [inline integrations/api_repo_test.go:408]: note: I haven't contributed API tests before so please guide me if the test looks awful.

File: routers/api/v1/repo/mirror.go [hunk 2]
```diff
@@ -48,6 +50,14 @@ func MirrorSync(ctx *context.APIContext) {
 		return
 	}
 
+	if _, err := repo_model.GetMirrorByRepoID(repo.ID); err != nil {
+		if errors.Is(err, repo_model.ErrMirrorNotExist) {
+			ctx.Error(http.StatusBadRequest, "MirrorSync", "Repository is not a mirror")
+			return
+		}
+		ctx.Error(http.StatusInternalServerError, "MirrorSync", err)
```
Comment (lunny) [inline routers/api/v1/repo/mirror.go:58]: return

Comment (petergardfjall) [reply routers/api/v1/repo/mirror.go:58]: 72c05e565

**Review by a1012112796 — APPROVED** [review]

**Review by lunny — APPROVED** [review]

File: integrations/api_repo_test.go (hunk 1 above)
Comment (techknowlogick) [reply integrations/api_repo_test.go:408]: Tests seem reasonable, and very much appreciated :)

**Pull request merged**